Create main.cpp

MaEgV opened the pull request:
Описание системы:
Есть абстрактный класс system, который подразумевает наличие метода отрисовки и метода взаимодействия с мышью
От него наследуются 2 класса : menu_launcher и editor_launcher. 
В модуле paint(главный модуль) регистрируются функции взаимодействия с glut. Если glut посылает сигнал нажатия кнопки, то модуль paint отправляет его в текущую систему (переменная grafic_system в классе paint). В зависимости от того, какая система выбрана, нажатие обрабатывается либо меню либо редактором.

Внутри редактора похожая схема:
Общий абстрактный класс cursor_mode, умеющий обрабатывать основные команды, поступающие из paint. От него наследуются подветки абстрактной кисти и примитивов. 
Выбор режима зависит от того, какой установлен режим курсора

MaEgV committed:
Create main.cpp

Описание системы:
Есть абстрактный класс system, который подразумевает наличие метода отрисовки и метода взаимодействия с мышью
От него наследуются 2 класса : menu_launcher и editor_launcher. 
В модуле paint(главный модуль) регистрируются функции взаимодействия с glut. Если glut посылает сигнал нажатия кнопки, то модуль paint отправляет его в текущую систему (переменная grafic_system в классе paint). В зависимости от того, какая система выбрана, нажатие обрабатывается либо меню либо редактором.

Внутри редактора похожая схема:
Общий абстрактный класс cursor_mode, умеющий обрабатывать основные команды, поступающие из paint. От него наследуются подветки абстрактной кисти и примитивов. 
Выбор режима зависит от того, какой установлен режим курсора

MaEgV committed:
Add files via upload

*Основной код в модулях paint и grafic_editor

Gandy123 commented on grafic_editor.cpp:
```diff
@@ -0,0 +1,257 @@
+bool lcm = false;
+bool ccm = false;
+bool rcm = false;
+
```
давайте как-то вменяемо их назовем

Gandy123 commented on grafic_editor.cpp:
```diff
@@ -0,0 +1,257 @@
+brush* brush_mode = new brush;
+eraser* eraser_mode = new eraser;
+polygon* polygon_mode = new polygon;
+line* line_mode = new line;
```
а почему это все не члены вашего класса ? editor ?

Gandy123 commented on grafic_editor.cpp:
```diff
@@ -0,0 +1,257 @@
+	return cmp1;
+}
+
+void polygon::draw(int x, int y) {
```
по модулям разбейте код пожалуйста, в идеале один класс два модуля cpp и h

Gandy123 reviewed (changes requested):
done

MaEgV committed:
Add files via upload

1) Глобальные переменные разместил в классах, которые их используют
2) Вытащил из "grafic_editor" все лишние классы, теперь там только подключаются непосредственно режимы отрисовки. Проверил, что все режимы работают автономно
3) Переименовал: lcm -> mouse_left, rcm -> mouse_right.

Gandy123 commented on cursore_mode.cpp:
```diff
@@ -0,0 +1,7 @@
+
+bool cursor_mode::lcm = false;
+bool cursor_mode::ccm = false;
+bool cursor_mode::rcm = false;
```
давайте это дело сделаем enum-ом и как-то переименуйте а то я не понимаю что они означают

MaEgV replied:
Я менял название этих переменных перед тем, как загружал исправления, и похоже забыл сохранить  :( 
Я писал об этом в комментарии к исправлениям

Это переменные отвечающие за состояние кнопок мыши, сейчас они называются:
lcm -> mouse_left, ccm->mouse_central, rcm -> mouse_right
В enum их нельзя превратить, потому что несколько кнопок могут быть нажаты одновременно

Gandy123 commented on eraser.cpp:
```diff
@@ -0,0 +1,6 @@
+#include"eraser.h"
+
+eraser::eraser() {
+	size = 4;
```
короткие методы лучше прям в h файле описывать, все что на одну строчку помещается

MaEgV replied:
Понял

Gandy123 commented on eraser.h:
```diff
@@ -0,0 +1,15 @@
+	eraser();
+
+	void set_color(color_type c) override {};
+
```
а зачем пустой метод ?

MaEgV replied:
В основном виртуальном классе отрисовки предусмотрена смена цвета для режима отрисовки. Стиралка меняет цвет пикселя на белый. Чтобы пользователь не мог менять на ней цвет, метод сделан пустым. А переопределен он, чтобы класс не был виртуальным.

Gandy123 reviewed (changes requested):
done

MaEgV committed:
Add files via upload
